app/views/admin/order_cycles/edit.html.haml

```diff
@@ -15,19 +15,46 @@
+      %input{ type: "button", "ng-value": "order_cycle_form.$dirty ? '#{t('.cancel')}' : '#{t('.back_to_list')}'", "ng-click": "cancel('#{main_app.admin_order_cycles_path}')" }
+    %div#modal-actions{style: "display: none;"}
+      %input.red{ type: "button", value: t('.save'), "ng-disabled": "!order_cycle_form.$dirty || order_cycle_form.$invalid", data: { 'action': 'click->modal-link#open click->order-cycle#updateModalConfirmButton', 'target': 'save'} }
+      - if @order_cycle.simple?
+        %input.red{ type: "button", value: t('.save_and_back_to_list'), "ng-disabled": "!order_cycle_form.$dirty || order_cycle_form.$invalid", data: { 'action': 'click->modal-link#open click->order-cycle#updateModalConfirmButton', 'target': 'saveAndBack'} }
+      - else
+        %input.red{ type: "button", value: t('.save_and_next'), "ng-disabled": "!order_cycle_form.$dirty || order_cycle_form.$invalid", data: { 'action': 'click->modal-link#open click->order-cycle#updateModalConfirmButton', 'target': 'saveAndNext'} }
+        %input{ type: "button", value: t('.next'), "ng-click": "cancel('#{main_app.admin_order_cycle_incoming_path(@order_cycle)}')", "ng-disabled": "order_cycle_form.$dirty" }
+      %input{ type: "button", "ng-value": "order_cycle_form.$dirty ? '#{t('.cancel')}' : '#{t('.back_to_list')}'", "ng-click": "cancel('#{main_app.admin_order_cycles_path}')" }
 
   - if @order_cycle.simple?
     = render 'simple_form', f: f
   - else
     = render 'form', f: f
+
+  - if has_scheduled_order
+    = render ModalComponent.new(id: "linked-schedule-warning-modal", close_button: false) do
+      .content
+        .modal-body
+          %h6
+            = t('admin.order_cycles.edit.linked_schedule_warning_modal.title')
+          %div{ style: 'font-size: 1rem;' }
+            = t('admin.order_cycles.edit.linked_schedule_warning_modal.content')
+        %p.modal-actions.justify-end
+          %button.button.secondary#modal-confirm{ type: "button", 'data-action': 'click->modal#close', style: 'display: none;', data: { 'order-cycle-target': 'modalConfirm', request: 'save' }, "ng-click": "submit($event, null)" }
+            = t('admin.order_cycles.edit.linked_schedule_warning_modal.proceed')
+          %button.button.secondary#modal-confirm{ type: "button", 'data-action': 'click->modal#close', style: 'display: none;', data: { 'order-cycle-target': 'modalConfirm', request: 'saveAndNext' }, "ng-click": "submit($event, '#{main_app.admin_order_cycle_incoming_path(@order_cycle)}')" }
+            = t('admin.order_cycles.edit.linked_schedule_warning_modal.proceed')
+          %button.button.secondary#modal-confirm{ type: "button", 'data-action': 'click->modal#close', style: 'display: none;', data: { 'order-cycle-target': 'modalConfirm', request: 'saveAndBack' }, "ng-click": "submit($event, '#{main_app.admin_order_cycles_path}')" }
+            = t('admin.order_cycles.edit.linked_schedule_warning_modal.proceed')
```

Comment:
I am not a big fan of having a mix of stimulus and angular actions on the same button, but can't really think of another way of doing it ?

Reply:
I agree, it doesn't look right, but not sure of a better way.  At least it's a way to continue the transition to Stimulus, away from Angular.